process/heartbeatProcessor.go

```diff
@@ -88,10 +88,11 @@ func (hbp *HeartbeatProcessor) getHeartbeatsFromApi() (*data.HeartbeatResponse,
 
 func (hbp *HeartbeatProcessor) addMessagesToMap(responseMap map[string]data.PubKeyHeartbeat, heartbeats []data.PubKeyHeartbeat, observerShard uint32) {
 	for _, heartbeatMessage := range heartbeats {
-		isMessageFromCurrentShard := heartbeatMessage.ReceivedShardID == observerShard
-		if !isMessageFromCurrentShard {
-			continue
-		}
+		// TODO: un-comment these lines when heartbeat v2 is merged on the node's latest release
```

Comment:
We need a proper solution. This comment will just push the same problem in the future

Reply:
Right now, it will maintain backward compatibility. In the future, we might return an error if a shard doesn't respond (see hyperblock's way)

Reply:
then let's rephrase the TODO, something like:
```GO
TODO: fix these merges when the heartbeat v2 will be active
```
otherwise, someone will just uncomment these and hope it will work

Reply:
updated todo